packages/core/src/middleware/koa-auth.ts

```diff
@@ -34,54 +34,58 @@ const extractBearerTokenFromHeaders = ({ authorization }: IncomingHttpHeaders) =
   return authorization.slice(bearerTokenIdentifier.length + 1);
 };
 
-type UserInfo = {
+type TokenInfo = {
   sub: string;
   roleNames?: string[];
 };
 
-const getUserInfoFromRequest = async (request: Request): Promise<UserInfo> => {
+export const verifyBearerTokenFromRequest = async (
+  request: Request,
+  resourceIndicator = managementResource.indicator
+): Promise<TokenInfo> => {
   const { isProduction, isIntegrationTest, developmentUserId, oidc } = envSet.values;
   const userId = request.headers['development-user-id']?.toString() ?? developmentUserId;
 
   if ((!isProduction || isIntegrationTest) && userId) {
     return { sub: userId, roleNames: [UserRole.Admin] };
   }
 
-  const { localJWKSet, issuer } = oidc;
-  const {
-    payload: { sub, role_names: roleNames },
-  } = await jwtVerify(extractBearerTokenFromHeaders(request.headers), localJWKSet, {
-    issuer,
-    audience: managementResource.indicator,
-  });
-
-  assertThat(sub, new RequestError({ code: 'auth.jwt_sub_missing', status: 401 }));
+  try {
+    const { localJWKSet, issuer } = oidc;
+    const {
+      payload: { sub, role_names: roleNames },
+    } = await jwtVerify(extractBearerTokenFromHeaders(request.headers), localJWKSet, {
+      issuer,
+      audience: resourceIndicator,
+    });
+
+    assertThat(sub, new RequestError({ code: 'auth.jwt_sub_missing', status: 401 }));
+
+    return { sub, roleNames: conditional(Array.isArray(roleNames) && roleNames) };
```

Comment:
nit: if `roleNames` is not an array, then it would end up with `roleNames: undefined`.

How about
```
return { sub, (...conditional(Array.isArray(roleNames) && { roleNames })) }
```

Reply:
actually we can use `null`

Reply:
discussed offline, keep as it is